torch/csrc/jit/tensorexpr/ir.h

```diff
@@ -485,22 +482,21 @@ class IfThenElse : public ExprNode<IfThenElse> {
       const ExprHandle& c,
       const ExprHandle& t,
       const ExprHandle& f) {
-    return ExprHandle(new IfThenElse(c.node(), t.node(), f.node()));
-  }
-
-  IfThenElse(const Expr* c, const Expr* t, const Expr* f)
-      : ExprNodeBase(t->dtype()), condition_(c), true_(t), false_(f) {
-    if (!c->dtype().is_integral()) {
+    if (!c.dtype().is_integral()) {
       throw unsupported_dtype();
     }
-    if (c->dtype().lanes() != 1) {
+    if (c.dtype().lanes() != 1) {
       throw unsupported_dtype();
     }
-    if (t->dtype() != f->dtype()) {
+    if (t.dtype() != f.dtype()) {
       throw malformed_input("Bad dtype in IfThenElse");
     }
```

Comment:
These have already been moved to the verifier, right? Why do we need them here as well?

Reply:
I kept them in `::make` methods whenever it was possible, since there they don't cause segfaults and will allow to catch the issue earlier.